ros/src/waypoint_updater/nodes/updater.py

```diff
@@ -0,0 +1,227 @@
+                         "waypoints when we have already loaded %d waypoints",
+                         len(self.waypoints))
+
+        self.subs['/base_waypoints'].unregister()
+        rospy.loginfo("Unregistered from /base_waypoints topic")
+
+    def traffic_cb(self, traffic_msg):
+        # TODO: Callback for /traffic_waypoint message. Implement
+        # self.next_traffic_light_wp = traffic_msg.data
+        pass
+
+    def obstacle_cb(self, msg):
+        # TODO: Callback for /obstacle_waypoint message.
+        # We will implement it later
+        pass
+
+    def send_waypoints(self):
+        # generates the list of LOOKAHEAD_WPS waypoints based on car location
+        # for now assume waypoints form a loop
+        new_wps_list = []
+        start_wps_ptr = self.closest_waypoint()
+        end_wps_ptr = (start_wps_ptr +
+                       self.lookahead_wps) % len(self.waypoints)
+        rospy.loginfo("waypoint_updater:send_waypoints start_wps_ptr = %d,"
+                      " end_wps_ptr = %d", start_wps_ptr, end_wps_ptr)
+        if end_wps_ptr > start_wps_ptr:
+            for w_p in self.waypoints[start_wps_ptr:end_wps_ptr]:
+                w_p.twist.twist.linear.x = self.default_velocity
+                new_wps_list.append(w_p)
+            # end of for
+        else:
+            for w_p in self.waypoints[start_wps_ptr:]:
+                w_p.twist.twist.linear.x = self.default_velocity
+                new_wps_list.append(w_p)
+            # end of for
+            for w_p in self.waypoints[:end_wps_ptr]:
+                w_p.twist.twist.linear.x = self.default_velocity
+                new_wps_list.append(w_p)
+            # end of for
+        # end of if
+        rospy.loginfo("waypoint_updater:send_waypoints final_waypoints list"
+                      " length is %d", len(new_wps_list))
+        lane = Lane()
+        lane.waypoints = list(new_wps_list)
+        lane.header.frame_id = '/world'
+        lane.header.stamp = rospy.Time()
+        self.pubs['/final_waypoints'].publish(lane)
+
+    def get_waypoint_velocity(self, waypoint):
+        return waypoint.twist.twist.linear.x
+
+    def set_waypoint_velocity(self, waypointlist, waypoint, velocity):
+        waypointlist[waypoint].twist.twist.linear.x = velocity
+
+    def closest_waypoint(self):
+        dist = 1000000  # maybe should use max
+        closest = 0
+
+        def distance_lambda(a, b): return math.sqrt(
+            (a.x-b.x)**2 + (a.y-b.y)**2)
+        for i in range(len(self.waypoints)):
+            tmpdist = distance_lambda(self.waypoints[i].pose.pose.position,
+                                      self.pose.position)
+            if tmpdist < dist:
+                closest = i
+                dist = tmpdist
+            # end of if
+        # end of for
+        return closest
+        # TODO - implement next_waypoint to make sure we start list at a point
+        # in front of the car
```

Comment:
After looking at the waypoint_follower logic, I don't think we need to do this anymore because I'm planning to modify it to not assume the first waypoint is ahead of the car.